### PR TITLE
Yoni - added the whole Pivot

### PR DESCRIPTION
**Pull request by bossyshrimp68**
No need to check this if you arent herman

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/FalconPivot.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import com.revrobotics.CANSparkMax;
+import edu.greenblitz.robotName.utils.motors.GBSparkMax;
+import edu.wpi.first.math.util.Units;
+
+import static edu.greenblitz.robotName.subsystems.Arm.Pivot.PivotConstants.*;
+import static edu.greenblitz.robotName.subsystems.Arm.Pivot.PivotConstants.FalconConfigs.*;
+
+public class FalconPivot implements IPivot{
+    private TalonFX motor;
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/FalconPivot.java:14]: space line

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/FalconPivot.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public FalconPivot() {
+        motor = new TalonFX(MOTOR_ID);
+        motor.getConfigurator().apply(MOTION_MAGIC_CONFIGS);
+        motor.getConfigurator().apply(CURRENT_LIMITS_CONFIGS);
+        motor.getConfigurator().apply(CLOSED_LOOP_RAMPS_CONFIGS);
+        motor.setNeutralMode(NEUTRAL_MODE_VALUE);
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/FalconPivot.java:20]: switch between this and line after this

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/FalconPivot.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Override
+    public void setPower(double power) {
+        motor.set(power);
+    }
+
+    @Override
+    public void moveToAngle(double goalAngle) {
+        motor.setControl(new MotionMagicDutyCycle(
+                goalAngle,
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/FalconPivot.java:38]: add conversion

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/FalconPivot.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Override
+    public void updateInputs(PivotInputsAutoLogged inputs) {
+        inputs.outputCurrent = motor.getSupplyCurrent().getValue();
+        inputs.appliedOutput = motor.getMotorVoltage().getValue();
+        inputs.position = motor.getPosition().getValue() * RELATIVE_POSITION_CONVERSION_FACTOR;
+        inputs.velocity = motor.getVelocity().getValue() * RELATIVE_VELOCITY_CONVERSION_FACTOR;
+        inputs.absoluteEncoderPosition = motor.getDutyCycle().getValue() * ABSOLUTE_POSITION_CONVERSION_FACTOR;
+        inputs.hasHitForwardLimit = false;
+        inputs.hasHitBackwardsLimit = false;
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/FalconPivot.java:67]: look at my code

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/IPivot.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    void setPower(double power);
+
+    void moveToAngle(double goalAngle);
+
+    void setVoltage(double voltage);
+
+    void setIdleMode(NeutralModeValue idleMode);
+
+
+    void updateInputs(PivotInputsAutoLogged inputs);
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/IPivot.java:18]: ctrl + alt + L

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/Pivot.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import edu.greenblitz.robotName.subsystems.Battery;
+import edu.greenblitz.robotName.utils.GBSubsystem;
+import edu.wpi.first.math.util.Units;
+import org.littletonrobotics.junction.Logger;
+
+import static edu.greenblitz.robotName.subsystems.Arm.Pivot.PivotConstants.FalconConfigs.*;
+
+public class Pivot extends GBSubsystem {
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/Pivot.java:11]: look at how i orginized the file

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/PivotInputs.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package edu.greenblitz.robotName.subsystems.Arm.Pivot;
+
+import org.littletonrobotics.junction.AutoLog;
+
+@AutoLog
+public class PivotInputs {
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/PivotInputs.java:6]: ctrl + alt + L

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/SimulationPivot.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import org.littletonrobotics.junction.Logger;
+
+import static edu.greenblitz.robotName.subsystems.Arm.Pivot.PivotConstants.Simulation.SIM_PID;
+
+public class SimulationPivot implements IPivot{
+    SingleJointedArmSim pivotSim;
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/SimulationPivot.java:15]: simulation

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/SimulationPivot.java [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    public void updateInputs(PivotInputsAutoLogged inputs) {
+
+        pivotSim.update(RobotConstants.SimulationConstants.TIME_STEP);
+
+        inputs.appliedOutput = appliedVoltage;
+        inputs.outputCurrent = pivotSim.getCurrentDrawAmps();
+        inputs.position = pivotSim.getAngleRads();
+        inputs. velocity = pivotSim.getVelocityRadPerSec();
+        inputs.absoluteEncoderPosition = pivotSim.getAngleRads();
+
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/SimulationPivot.java:78]: why space

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/Pivot.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        pivot.setVoltage(voltage);
+    }
+
+    public void setIdleMode(NeutralModeValue idleMode) {
+        pivot.setIdleMode(idleMode);
+    }
+    public double getVoltage() {
+        return pivotInputs.appliedOutput * Battery.getInstance().getCurrentVoltage();
+    }
+    public double getVelocity() {
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/Pivot.java:59]: velocity in what half deegres?

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/Pivot.java (hunk 6 above)
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/Pivot.java:11]: add get angle

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/SimulationPivot.java [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+import edu.greenblitz.robotName.RobotConstants;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import org.littletonrobotics.junction.Logger;
+
+import static edu.greenblitz.robotName.subsystems.Arm.Pivot.PivotConstants.Simulation.SIM_PID;
+
+public class SimulationPivot implements IPivot{
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/SimulationPivot.java:14]: ctrl alt l

File: src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/Pivot.java (hunk 6 above)
Comment (bossyshrimp68) [reply src/main/java/edu/greenblitz/robotName/subsystems/Arm/Pivot/Pivot.java:11]: thumbs down emojo

**Pull request merged**